File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.java
class RandomizedSet {
    private Set<Integer> set;
    public RandomizedSet() {
        set = new HashSet<>();
    }
    
    public boolean insert(int val) {
        boolean exists = set.contains(val); 
        if (exists) return false;
        set.add(val);
        return true;
    }
    
    public boolean remove(int val) {
        boolean exists = set.contains(val);
        if (exists) set.remove(val);
        return exists;
    }
    
    public int getRandom() {
        List<Integer> list = new ArrayList<>(set);
        Random random = new Random();
        return list.get(random.nextInt(list.size()));
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.kt
class RandomizedSet() {
    private val set = HashSet<Int>()
    fun insert(`val`: Int): Boolean {
        val exists = set.contains(`val`);
        set.add(`val`)
        return !exists;
    }

    fun remove(`val`: Int): Boolean {
        val exists = set.contains(`val`);
        set.remove(`val`)
        return exists;
    }

    fun getRandom(): Int {
        val list = ArrayList(set)
        return list[kotlin.random.Random.nextInt(list.size)]
    }

}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * var obj = RandomizedSet()
 * var param_1 = obj.insert(`val`)
 * var param_2 = obj.remove(`val`)
 * var param_3 = obj.getRandom()
 */